rocAL/source/pipeline/pipeline_serializer.cpp

```
@@ -0,0 +1,186 @@
+        THROW("Failed to open file for writing serialized pipeline: " + file_path);
+    }
+    if (!_pipeline_proto.SerializeToOstream(&ofs)) {
+        THROW("Failed to serialize pipeline to file: " + file_path);
+    }
+}
+
+void PipelineSerializer::serialize_pipeline_config(size_t num_threads, size_t batch_size, int device_id, RocalMemType device_type, size_t prefetch_queue_depth) {
+    _pipeline_proto.set_num_threads(num_threads);
+    _pipeline_proto.set_batch_size(batch_size);
+    _pipeline_proto.set_device_id(device_id);
+    _pipeline_proto.set_rocal_cpu(device_type == RocalMemType::HOST ? true : false);
+    _pipeline_proto.set_prefetch_queue_depth(prefetch_queue_depth);
+}
+
+void set_tensor_proto(rocal_proto::InputOutput *in_out_proto, Tensor *tensor, bool is_input = false) {
+    in_out_proto->set_name(tensor->tensor_name());
+    in_out_proto->set_device(static_cast<int>(tensor->info().mem_type()));
+    in_out_proto->set_dtype(static_cast<int>(tensor->info().data_type()));
+    in_out_proto->set_layout(static_cast<int>(tensor->info().layout()));
+    in_out_proto->set_color_format(static_cast<int>(tensor->info().color_format()));
+    for (auto &dim : tensor->info().dims())
+        in_out_proto->add_dims(dim);
+    in_out_proto->set_num_dims(tensor->info().num_of_dims());
+    in_out_proto->set_is_argument_input(is_input);
+}
+
+void PipelineSerializer::serialize_pipeop_arguments(const std::vector<Argument>& arguments_list, rocal_proto::OperatorDef *opdef) {
+
+    // Iterate through each argument to store in the protobuffers
+    for (auto &op_arg : arguments_list) {
+        rocal_proto::Arguments *arg = opdef->add_args();
+        arg->set_name(op_arg.arg_name);
+        arg->set_type(op_arg.type_name);
+        arg->set_is_vector(op_arg.is_vector);
+
+        if (op_arg.type_name == "nullptr") continue;
+        if (op_arg.sub_type_name != "")
+            arg->set_instance_name(op_arg.sub_type_name);
+
+        if (op_arg.is_parameter) {
```

Many operators pass `FloatParam*`/`IntParam*` arguments; hitting this branch means we currently emit nothing for them. The serialized pipeline will therefore lose all randomized/parameterized configuration. Please either serialize the parameter payload here or fail fast until support lands so consumers aren’t left with incomplete exports.